src/kfactory/kcell.py

```diff
@@ -3916,7 +3919,15 @@ def create_port(
                 kcl=self.kcl,
             )
         elif dcplx_trans is not None:
-            assert dwidth is not None
+            if dwidth is None or dwidth <= 0:
+                raise ValueError("dwidth needs to be set")
+            elif dwidth is not None and dwidth > 0:
+                width = round(dwidth / self.kcl.dbu)
+                width = cast(int, width)
```

Comment:
I believe this is not necessary. This only has an impact on the type checking. width should already be an integer due to the `round`. Does mypy fail without this? Also afaik this might wipe width because cast does a nop in no-type-checking context